### DISCUSSION TITLE
Allow to set RDP app and parameters on Windows (and Mac) clients

### DISCUSSION THREAD

**Discussion by vieira-temes** [rustdesk/rustdesk#2017] [General]
RDP can be used on Windows via two applications:

- The good old Remote Desktop Connection, which is part of Windows
- Microsoft Remote Desktop, which can be installed from the Microsoft Store (or the AppStore in Mac computers)

Each has its own advantages.

The main advantage for RDC is it can handle multiple monitors.
The best feature of MRD is it will allow resizing the remote desktop to the client's windows size to always be pixel-perfect at the same DPI as the client and with no scroll bars.

At the moment, Rustdesk seems to be using RDC, but with hardcoded parameters. I, for one, can't use it because my laptop has a very HiDPI resolution and on the remote host everything looks tiny, the window is tiny too and it's basically useless as it is now.

I would like to suggest Rustdesk to allow the user to configure RDP to use either RDC or MRD and set parameters for each protocol. I don't think it would be necessary to have parameter settings per remote host. Global ones would be fine. I think taking inspiration from the current Microsoft apps, (especially Microsoft Remote Desktop, which also works on Macs), would be a good idea.

**Comment by rustdesk** [comment]
pr welcome

**Comment by vieira-temes** [comment]
What's the meaning of "pr"?

**Comment by kolAflash** [comment]
@PabloGrok  
pr = https://en.wikipedia.org/wiki/Distributed_version_control#Pull_requests

By this @rustdesk asks, if you can provide the source code to implement this. So he can "pull" the nessecary changes from your source code.

**Comment by vieira-temes** [comment]
Oh, OK, thanks for the explanation.

The developer shouldn't assume everyone has development skills, but I understand he's in a hurry to reply to posts.

This was just a suggestion to improve the product.

**Comment by ChaosFenix** [comment]
I would just like to add that I would also like for this to be possible. I am not sure if it would be possible to integrate FreeRDP or something similar in order to implement it. Rustdesk can already determine if a host is on the same network through the built in discovery. It would be really nice if RDP was an allowed connection type. Bonus points if RDP could be specified as the primary connection type and then falling back to the framebuffer if the host doesn't support an rdp connection or if it isn't on the same network. Like Pablo I am not a programmer and so this is simply a feature request and not a pull request.
